Facebooky/CheckNetworkLink.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Net.NetworkInformation;
+using System.Windows.Forms;
+
+using FacebookWrapper.ObjectModel;
+
+namespace Facebooky
+{
+	class CheckNetworkLink : IPostChainLink
+	{
+		public IPostChainLink NextLink { get; set; }
+		public bool Enabled { get; set; }
+		public User LoggedInUser { get; set; }
+		public void HandlePost(PostInfo i_Post)
+		{
+			SendPostCommand postCommand = new SendPostCommand();
+			postCommand.Post = i_Post;
+			postCommand.LoggedInUser = LoggedInUser;
+			bool networkAvalable = System.Net.NetworkInformation.NetworkInterface.GetIsNetworkAvailable();
+			if (Enabled && !networkAvalable)
+			{
+				System.Net.NetworkInformation.NetworkChange.NetworkAvailabilityChanged += (i_Sender, i_Args) => { postCommand.Execute(); };
```

Comment:
We already discussed the fact that you can't use a lambda expression because you will need to remove yourself from the event after being executed.